src/streams/adapters/EventedStreamSource.ts

```diff
@@ -0,0 +1,50 @@
+import Evented from '../../Evented';
+import { Handle } from '../../interfaces';
+import on, { EventEmitter, EventTarget, ExtensionEvent } from '../../on';
+import Promise from '../../Promise';
+import { Source } from '../ReadableStream';
+import ReadableStreamController from '../ReadableStreamController';
+
+type EventTargetTypes = Evented | EventEmitter | EventTarget;
+type EventTypes = Array<ExtensionEvent | string> | ExtensionEvent | string;
+
+export default class EventedStreamSource implements Source<Event> {
+	protected _controller: ReadableStreamController<Event>;
+	protected _target: EventTargetTypes;
+	protected _events: Array<ExtensionEvent | string>;
+	protected _handles: Handle[];
+
+	constructor(target: EventTargetTypes, type: EventTypes) {
+		this._target = target;
+
+		if (Array.isArray(type)) {
+			this._events = <any> type;
+		}
+		else {
+			this._events = [ <any> type ];
```

Comment:
I think what we have agreed is that we should at least comment why we have an `any` cast, because the intent isn't always (seldom) clear.